src/Request.php

```diff
@@ -53,12 +53,11 @@ public function connect()
                 $this->getConnectOption('login_method', 'AMQPLAIN'),
                 $this->getConnectOption('login_response', null),
                 $this->getConnectOption('locale', 3),
+                $this->getConnectOption('channel_rpc_timeout', 0.0),
                 $this->getConnectOption('read_write_timeout', 130),
                 $this->getConnectOption('context', null),
                 $this->getConnectOption('keepalive', false),
-                $this->getConnectOption('heartbeat', 60),
```

Comment:
Why would you delete those?
`$channel_rpc_timeout = 0.0,
        $ssl_protocol = null`
We should still keep both connect options configurable.

Reply:
I am fine with keeping both those options configurable, I was simply removing them as they were unnecessary and out of order for the current AMQPStreamConnection constructor. Adding them back is fine by me.

The next question is what do you want the 16th parameter to be called in the options as even with them here we the StreamConnection is only passing in 15 of the 16 parameters (it looks like the connection_timeout parameter is the one missing) (and good lord, that is way too many parameters in either case, but that is a whole other discussion). Here's a copy of the ctor's declaration for reference:
```php
    public function __construct(
        $host,
        $port,
        $user,
        $password,
        $vhost = '/',
        $insist = false,
        $login_method = 'AMQPLAIN',
        $login_response = null,
        $locale = 'en_US',
        $connection_timeout = 3.0,
        $read_write_timeout = 130.0,
        $context = null,
        $keepalive = false,
        $heartbeat = 60,
        $channel_rpc_timeout = 0.0,
        $ssl_protocol = null
    ) {
```

Reply:
As far as I can see they are all used. By SteamIO or by passing into the AbstractConnection constructor.

Name should be consistent with the underlying implementation.

I agree, we will not fix the amqplib ctor argument count in this PR. ;-)

Reply:
Sounds good. I have made the updates for all comments in this PR thus far.